gen/enum_test.go

```diff
@@ -399,3 +399,69 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumLabelLegit(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithLabel
+		wireValue string
+	}{
+		{te.EnumWithLabelUsername, "\"surname\""},
+		{te.EnumWithLabelPassword, "\"hashed_password\""},
+		{te.EnumWithLabelSalt, "\"salt\""},
+		{te.EnumWithLabelSugar, "\"sugar\""},
+	}
+	for _, tt := range tests {
+		t.Run("compare label:"+tt.wireValue, func(t *testing.T) {
+			// marshal
```

Comment:
added marshal / unmarshal behavior

Reply:
Can we put the JSON behavior check in a subtest since this is also testing roundtripping.

```
t.Run("JSON round trip", func(t *testing.T) {
  ...
})
assertRoundTrip(...)
```